datafusion/functions/src/crypto/digest.rs

```diff
@@ -19,13 +19,38 @@
 use super::basic::{digest, utf8_or_binary_to_binary_type};
 use arrow::datatypes::DataType;
 use datafusion_common::Result;
-use datafusion_expr::scalar_doc_sections::DOC_SECTION_HASHING;
 use datafusion_expr::{
     ColumnarValue, Documentation, ScalarUDFImpl, Signature, TypeSignature::*, Volatility,
 };
+use datafusion_macros::user_doc;
 use std::any::Any;
-use std::sync::OnceLock;
 
+#[user_doc(
+    doc_section(label = "Hashing Functions"),
+    description = "Computes the binary hash of an expression using the specified algorithm.",
+    syntax_example = "digest(expression, algorithm)",
+    sql_example = r#"```sql
+> select digest('foo', 'sha256');
++------------------------------------------+
+| digest(Utf8("foo"), Utf8("sha256"))      |
++------------------------------------------+
+| <binary_hash_result>                     |
++------------------------------------------+
+```"#,
+    standard_argument(name = "expression", prefix = "String"),
+    argument(
+        name = "algorithm",
+        description = "String expression specifying algorithm to use. Must be one of:       
+- md5
+- sha224
+- sha256
+- sha384
+- sha512
+- blake2s
+- blake2b
+- blake3"
```

Comment:
```suggestion
    - md5
    - sha224
    - sha256
    - sha384
    - sha512
    - blake2s
    - blake2b
    - blake3"
```
Maybe we can add some indents for them to format them like
![Screenshot 2024-12-28 at 11 38 10 AM](https://github.com/user-attachments/assets/aeaa9975-3db6-4fc6-b7c7-677fabb8e951)